### ISSUE TITLE
[Java] FlightStream: how to check whether data is available in the stream in a nonblocking way; 

### ISSUE THREAD

**Issue by bmmcq** [apache/arrow#34600]
### Describe the usage question you have. Please include as many useful details as  possible.

as the title: 
How to check whether data is available in the stream in a nonblocking way; 

It seems that the FlightStream API only supports fetching data by calling the `next()`  method in blocking; I think a nonblocking method like `tryNext()` should also be needed. 

### Component(s)

Java

**Comment by lidavidm** [comment]
Correct, there's not an async API. This has been discussed several times, but it would be a good amount of work to implement. #34221, #21039 are similarish issues - maybe I should file an overarching tracking issue.

**Comment by lidavidm** [comment]
I filed #34607 to try to centralize discussion about async APIs.

That said, it may be possible to add just a `tryNext` as a simple workaround in this case.
